### DISCUSSION TITLE
How to change (or set) connection type of I/O Messaging?

### DISCUSSION THREAD

**Discussion by Chungwei5** [ottowayi/pycomm3#288] [Q&A] [answered]
I'm using CIPDriver to communicate with an IFM IO-Link master (AL2301).
As information from internet there're different connection types: Exclusive Owner, Input-Only, Listen-Only.
To do some setup I need to write a output assembly only in excusive mode.
If the AL2301 is connected to PLC I can do the setup by choosing the 'Exclusive Owner' connection type.
But when using CIPDriver I don't know what connection type it is. And the write command has no error response but nothing is modified in the IO-link master.
Anyone know what connection type is when using the CIPDriver? And how can I change to 'Exclusive Owner' connection type?
Thanks a lot!

**Comment by ottowayi** [comment, marked as answer]
Unfortunately I don't think that is supported by the driver currently, I think those connection types refer to the class 0/1 connections.  Those are I/O connections and this library uses class 3, which is like the MSG instruction or HMI/SCADA connection. It may work by messing around with the forward open parameters, but currently those are not configurable or user facing.

**Reply by Chungwei5** [reply]
Understood. Thank you very much!
